packages/valory/skills/oracle_deployment_abci/models.py

```diff
@@ -35,23 +38,36 @@
 class SharedState(BaseSharedState):
     """Keep the current shared state of the skill."""
 
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        """Initialize the state."""
-        super().__init__(*args, abci_app_cls=OracleDeploymentAbciApp, **kwargs)
+    abci_app_cls = OracleDeploymentAbciApp
 
 
 class Params(BaseParams):
     """Parameters."""
 
-    observation_interval: float
-
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         """Initialize the parameters object."""
-        self.oracle_params = self._ensure("oracle", kwargs)
+        self.oracle_params = self._ensure("oracle", kwargs, Dict[str, Union[int, str]])
         super().__init__(*args, **kwargs)
 
 
-class RandomnessApi(ApiSpecs):
+# TODO remove this workaround when the types get fixed in `open-autonomy`
+class FixedApiSpecs(ApiSpecs):
+    """A model that wraps APIs to get cryptocurrency prices."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize ApiSpecsModel."""
+        self.url: str = self._ensure("url", kwargs, str)
+        self.api_id: str = self._ensure("api_id", kwargs, str)
+        self.method: str = self._ensure("method", kwargs, str)
+        self.headers = kwargs.pop("headers", [])
+        self.parameters = kwargs.pop("parameters", [])
+        self.response_info = ResponseInfo.from_json_dict(kwargs)
+        self.retries_info = RetriesInfo.from_json_dict(kwargs)
+        super(Model, self).__init__(*args, **kwargs)  # pylint: disable=bad-super-call
+        self._frozen = True
```

Comment:
The `ApiSpec`'s `headers` and `parameters` are optional parameters but are now using `_ensure` instead of `kwargs.pop` in the `open-autonomy`, which should not be the case to allow for empty values.

Reply:
All default values need to be defined on the skill.yaml. That's the intentional design. Otherwise they cannot be overridden on the agent.

Reply:
Fixed on 62b1b98ebb0a25134cad4632cac54eacd9f54549. However, the workaround is still required because the types are incorrect.